### PR TITLE
[EXD-88] Added "description" field to Canvas & simple search by part …

### PR DESCRIPTION
**Pull request by marcin-piela-ssh**
…of description (case insensitive)

### CODE REVIEW

File: src/canvas/canvas.interface.ts [hunk 1]
```diff
@@ -68,12 +70,16 @@ export class CanvasMetadataUpdateDTO {
   @MinLength(3)
   @MaxLength(255)
   name: string;
+  @MaxLength(4000)
+  description: string;
```
Comment (coredumped7893) [inline src/canvas/canvas.interface.ts]: description should be optional field, currently we will get http 400 on create and update endpoints

```json
{
    "message": [
        "description must be shorter than or equal to 4000 characters"
    ],
    "error": "Bad Request",
    "statusCode": 400
}
```

File: src/canvas/canvas.interface.ts [hunk 2]
```diff
@@ -68,12 +70,16 @@ export class CanvasMetadataUpdateDTO {
...
 }
 
 export class CanvasCreateDTO {
   @MinLength(3)
   @MaxLength(255)
   name: string;
+  @MaxLength(4000)
+  description: string;
```
Comment (coredumped7893) [inline src/canvas/canvas.interface.ts]: description should be optional field, currently we will get http 400 on create and update endpoints

**Review by coredumped7893 — CHANGES_REQUESTED** [review]

File: src/canvas/canvas.interface.ts (hunk 1 above)
Comment (marcin-piela-ssh) [reply src/canvas/canvas.interface.ts]: @coredumped7893 Done

File: src/canvas/canvas.interface.ts (hunk 2 above)
Comment (marcin-piela-ssh) [reply src/canvas/canvas.interface.ts]: @coredumped7893 Done

**Review by coredumped7893 — APPROVED** [review]

**Pull request merged**